tests/vtpm/conftest.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+@pytest.fixture(scope='module')
+def uefi_unix_halted_vm(uefi_vm, unix_vm, halted_vm):
+    yield uefi_vm
+
+@pytest.fixture(scope='module')
+def unix_vm_with_vtpm(uefi_unix_halted_vm):
+    vm = uefi_unix_halted_vm
+    vtpm_uuid = None
+
+    if vm.get_vtpm() is None:
+        vtpm_uuid = vm.create_vtpm()
+
+    snapshot = vm.snapshot()
```

Comment:
It doesn't really matter since destroying a snapshot doesn't destroy the associated vtpm that would have been created after the snapshot. You have to destroy the vtpm anyway.

And it's even worth because if you do:
```
snapshot = vm.snapshot()
vm.create_vtpm()
<...>
snapshot.destroy()
```
Then the vtpm will still exist and be attached to the VM but won't be usable by the VM anymore...

Reply:
And yes, there's a missing `snapshot.revert()`